torchgeo/samplers/single.py

```diff
@@ -142,10 +142,9 @@ def __iter__(self) -> Iterator[BoundingBox]:
             hit = self.hits[idx]
             bounds = BoundingBox(*hit.bounds)
 
-            # Choose a random index within that tile
-            bounding_box = get_random_bounding_box(bounds, self.size, self.res)
-
-            yield bounding_box
+            yield get_random_bounding_box_check_valid_overlap(
```

Comment:
maybe first add it as an option in the API, and pass the max_retries too

Reply:
Pushing a suggestion now